pkg/volume/glusterfs/glusterfs.go

```diff
@@ -288,23 +288,51 @@ func (c *glusterfsUnmounter) TearDownAt(dir string) error {
 
 func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 	var errs error
-
 	options := []string{}
+	hasLogFile := false
+	log := ""
+
 	if b.readOnly {
 		options = append(options, "ro")
+
 	}
 
-	p := path.Join(b.glusterfs.plugin.host.GetPluginDir(glusterfsPluginName), b.glusterfs.volName)
-	if err := os.MkdirAll(p, 0750); err != nil {
-		return fmt.Errorf("failed to create directory %v: %v", p, err)
+	// Check logfile has been provided by user, if provided, use that as the log file.
+	for i, userOpt := range b.mountOptions {
+		if dstrings.Contains(userOpt, "log-file") {
+			glog.V(4).Infof("log-file mount option is provided")
+			hasLogFile = true
+			logOptions := dstrings.Split(userOpt, "=")
+
+			if len(logOptions) > 1 && logOptions[1] != "" {
+				log = logOptions[1]
+			} else {
+				glog.Errorf("log-file location provided is nil, log-file location will be decided by the plugin")
+				b.mountOptions = append(b.mountOptions[:i], b.mountOptions[i+1:]...)
+				hasLogFile = false
+			}
+
+		}
 	}
 
-	// adding log-level ERROR to remove noise
-	// and more specific log path so each pod has
-	// its own log based on PV + Pod
-	log := path.Join(p, b.pod.Name+"-glusterfs.log")
-	options = append(options, "log-level=ERROR")
+	// If logfile has not been provided, create driver specific log file.
+	if !hasLogFile {
+		log = ""
+		p := path.Join(b.glusterfs.plugin.host.GetPluginDir(glusterfsPluginName), b.glusterfs.volName)
+		if err := os.MkdirAll(p, 0750); err != nil {
+			return fmt.Errorf("failed to create directory %v: %v", p, err)
+		}
+
+		// adding log-level ERROR to remove noise
+		// and more specific log path so each pod has
+		// its own log based on PV + Pod
+		log = path.Join(p, b.pod.Name+"-glusterfs.log")
+
+	}
+
+	// Use derived/provided log file in gluster fuse mount
 	options = append(options, "log-file="+log)
+	options = append(options, "log-level=ERROR")
```

Comment:
What if someone wants also custom log level, e.g. to debug something?

Reply:
true, possible and I was thinking to address it in another PR, Is it fine ? or do I need to address `log-level` as well in this PR ?

Reply:
The main reason for  a validation of `=` or `empty` was to fall back to default or existing way of creating log file. I dont know, failing if log-file option not provided properly or falling back to default log-file is preferred. I was thinking latter is better, but look like you prefer to fail if option is not provided properly. Please confirm.

Reply:
> I was thinking to address it in another PR, Is it fine ?

Yes, it's fine.

Reply:
Thanks!